libraries/AC_AttitudeControl/AC_WeatherVane.cpp

```diff
@@ -122,8 +122,9 @@ bool AC_WeatherVane::get_yaw_out(float &yaw_output, const int16_t pilot_yaw, con
     if (is_landing && (_landing_direction >= 0)) {
         dir = (Direction)_landing_direction.get();
     }
-    if (dir == Direction::OFF) {
+    if (dir == Direction::OFF || (dir == Direction::TAKEOFF_OR_LAND_ONLY && !(is_takeoff || is_landing))) {
```

Comment:
```suggestion
    if (dir == Direction::OFF || (dir == Direction::TAKEOFF_OR_LAND_ONLY) {
```
You don't need the extra bit because if take off or landing the two if's above will have changed dir.

Reply:
agreed